pytest_ethereum/linker.py

```diff
@@ -24,21 +25,28 @@ def _linker(operations: Callable[..., Any], package: Package) -> Callable[..., P
     return pipe(package, *operations)
 
 
-def deploy(contract_name: str, *args: Any) -> Callable[..., Tuple[Package, Address]]:
+def deploy(
+    contract_name: str, *args: Any, **kwargs: Any
+) -> Callable[..., Tuple[Package, Address]]:
     """
     Return a newly created package and contract address.
     Will deploy the given contract_name, if data exists in package. If
     a deployment is found on the current w3 instance, it will return that deployment
     rather than creating a new instance.
     """
-    return _deploy(contract_name, args)
+    return _deploy(contract_name, args, kwargs)
 
 
 @cytoolz.curry
-def _deploy(contract_name: str, args: Any, package: Package) -> Tuple[Package, Address]:
-    deployments = package.deployments
-    if contract_name in deployments:
-        return package, package.deployments[contract_name].address
+def _deploy(
+    contract_name: str, args: Any, kwargs: Any, package: Package
+) -> Tuple[Package, Address]:
+    # Catch no matching chain error from package.deployments
+    try:
+        if contract_name in package.deployments:
+            return package, package.deployments[contract_name].address
```

Comment:
I'm just realizing that my *gut* is telling me this lazy caching might be a bad idea.

Suppose my contract is stateful.  If in a test I end up calling this twice to deploy two different versions of the same contract, I would actually only end up with two instances representing the same contract.  Maybe this should be removed in favor of just always re-deploying and we can come up with a strategy for re-using deployments later.

Reply:
cool, yeah that's a good point and it was already causing some problems re the comment on line 44, so i'm happy to drop it.